main.go

```diff
@@ -279,18 +280,24 @@ func main() {
 
 	//+kubebuilder:scaffold:builder
 
-	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
+	if err = mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
 		setupLog.Error(err, "unable to set up health check")
 		os.Exit(1)
 	}
-	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
+	if err = mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
 		setupLog.Error(err, "unable to set up ready check")
 		os.Exit(1)
 	}
 
-	setupLog.Info("starting manager")
-	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
```

Comment:
Why remove this `mgr.Start(ctrl.SetupSignalHandler())` ? It's not the responsibility of a controller to do that.

Reply:
Because now the `stateOfheWorld` controller will start the manager: https://github.com/Kuadrant/policy-machinery/blob/85837a3f6a5ae4b84594de2124c5338f678a57f3/controller/controller.go#L176

> It's not the responsibility of a controller to do that.

The the responsibility of a Policy Machinery controller to do that.

Reply:
I do not think it is the responsibility of a policy machinery controller to do that. What if I am a that library user that wants to use that kuadrant's policy machinery together with another controller or  what if I want to manage the start of the manager myself?

Reply:
It's by design @eguzki. If you want to use a Policy Machinery controller, the manager associated with this controller will be started when you start the controller.

Keep in mind that the proposed pattern here is of the Policy Machinery controller being the one and only controller. One who feels the need for more controllers aside is probably not making enough use of Policy Machinery's reconcile funcs and patterns. Using a `Workflow` with more tasks instead maybe a good idea, for example.

But if you are "that user" who wants to break the pattern by having other controllers on the side, decoupling the starting of the manager, then I'm guessing your two sets of controllers (one set being actually a set of Policy Machinery reconciliation "tasks" and the other set being the collection other kinds of custom controllers not based on the Policy Machinery) are for completely independent "reconciliation realms". In this case, I would recommend creating two completely separate managers then. You can share the same context returned by `ctrl.SetupSignalHandler()` in both if you want. Start the Policy Machinery-based controller in a goroutine and then the other manager blocking the process.

IMHO, thinking in terms of "reconciliation realms" (as opposed to all things are interconnected) has been the core problem we want to fix with this work in the Kuadrant Operator.